app/src/main/java/com/example/tasky/feature_agenda/presentation/task_detail_screen/TaskDetailViewModel.kt

```diff
@@ -0,0 +1,141 @@
+import com.example.tasky.feature_agenda.domain.use_case.AgendaUseCases
+import com.example.tasky.feature_agenda.domain.util.ReminderType
+import com.example.tasky.util.Resource
+import com.example.tasky.util.Result
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import java.time.ZoneId
+import java.time.ZonedDateTime
+import java.util.UUID
+import javax.inject.Inject
+
+@HiltViewModel
+class TaskDetailViewModel @Inject constructor(
+    private val useCases: AgendaUseCases,
+    private val repositories: AgendaRepositories
+): ViewModel() {
+
+    private val _state = MutableStateFlow(TaskDetailState())
+    val state = _state.asStateFlow()
+
+    private val resultChannel = Channel<Result<Unit>>()
+    val logoutResult = resultChannel.receiveAsFlow()
+
+    fun onEvent(event: TaskDetailEvent) {
+        when(event) {
+            is TaskDetailEvent.AtDateChanged -> {
+                _state.update {
+                    it.copy(atDate = event.date)
+                }
+            }
+            is TaskDetailEvent.AtTimeChanged -> {
+                _state.update {
+                    it.copy(atTime = event.time)
+                }
+            }
+            TaskDetailEvent.CreateTask -> {
+                createTask()
+            }
+            is TaskDetailEvent.DescriptionChanged -> {
+                _state.update {
+                    it.copy(taskDescription = event.description)
+                }
+            }
+            is TaskDetailEvent.ReminderTypeChanged -> {
+                _state.update {
+                    it.copy(reminderType = event.reminderType)
+                }
+            }
+            is TaskDetailEvent.TitleChanged -> {
+                _state.update {
+                    it.copy(taskTitle = event.title)
+                }
+            }
+            TaskDetailEvent.ToggleEditMode -> {
+                _state.update {
+                    it.copy(editMode = !state.value.editMode)
+                }
+            }
+
+            is TaskDetailEvent.GetOpenedTask -> {
+                getSelectedTask(event.taskId)
+            }
+
+            is TaskDetailEvent.DeleteTask -> {
+                val taskId = event.taskId
+                taskId?.let {
+                    viewModelScope.launch {
+                        val result = repositories.taskRepository.getTask(it)
+                        when(result) {
+                            is Resource.Error -> resultChannel.send(Result.Error(result.message ?: "Unknown Error"))
+                            is Resource.Success -> {
+                                val task = result.data ?: return@launch
+                                repositories.taskRepository.deleteTask(task)
```

Comment:
Here, you'd save one DB call if you'd delete the task by its ID, since you already have that in this VM